### ISSUE TITLE
Duplicates <table> entry in clipboard when copying multiple paragraphs in a <td>

### ISSUE THREAD

**Issue by kg782** [ProseMirror/prosemirror#1270]
When copying multiple paragraphs in a table cell, it copies from `<table>` entry and also put another `<table>` in `data-pm-slice`. It causes unexpected behaviour when pasting.

Our product is: [https://atlaskit.atlassian.com/examples/editor/editor-core/full-page](https://atlaskit.atlassian.com/examples/editor/editor-core/full-page)

### Steps to reproduce

1. Access [ this CodeSandbox](https://codesandbox.io/s/dreamy-darkness-tjjpme?file=/src/index.js). I’m using the Chrome 101 on Mac 12.3

2. Select multiple paragraphs (not entire table cell)
<img width="178" alt="Screen Shot 2022-05-12 at 2 15 43 pm" src="https://user-images.githubusercontent.com/519868/167998223-098a7afd-6308-4d67-ade0-729fda64dde1.png">

3. Copy

4. Check the clipboard content with [Clipboard Inspector](https://nathf.github.io/clipboard-inspector/)  

### Expected vs Actual

**Actual**

It copies `<table>` and also put another table entry in `data-pm-slice`.

<img width="1011" alt="Screen Shot 2022-05-12 at 2 25 14 pm" src="https://user-images.githubusercontent.com/519868/167998593-5109af0f-3032-4ca2-be05-ce914085219d.png">

**Expected**

(I think) it should copy only multiple `<p>`s then `<table>` should be inserted to `data-pm-slice` which is the same behaviour as single line copy in a table cell.

<img width="720" alt="Screen Shot 2022-05-12 at 2 24 49 pm" src="https://user-images.githubusercontent.com/519868/167998613-dec5d031-87e4-4c0e-9e4a-a357ee75d81e.png">

### Possible solution

In my observation, it was caused by the last condition here.

https://github.com/ProseMirror/prosemirror-view/blob/master/src/clipboard.js#L6

When `<td>` has multiple children, it skips putting the `<td>` in `data-pm-slice`, put only `<table>` and `<tr>` there and leave the `<td>`. Then following `needsWrap` logic adds up another `<table>` entry then it is duplicated.

My suggesting solution is

[https://github.com/ProseMirror/prosemirror-view/compare/master...kg782:fix-clipboard](https://github.com/ProseMirror/prosemirror-view/compare/master...kg782:fix-clipboard)

Please give me your opinion.

**Comment by marijnh** [comment]
Does a patch that sets `context` to `[]` at the start of the second loop (`while (firstChild && ...)`) also solve the problem for you? That seems less likely to cause complications.

**Comment by kg782** [comment]
@marijnh Thank you for your advice. Yes, I confirmed the way fixes the problem and passed all your/our tests then made a PR.

https://github.com/ProseMirror/prosemirror-view/pull/124

**Comment by kg782** [comment]
I realised a problem with [the approach](https://github.com/ProseMirror/prosemirror-view/pull/124). It's removing genuine table which may have important attributes then keeping empty table. Closed the PR.

[This approach](https://github.com/ProseMirror/prosemirror-view/compare/master...kg782:fix-clipboard) might work better.

**Comment by marijnh** [comment]
Good point. Attached patch implements yet another approach, which should solve this in a more straightforward way.
